### PR TITLE
unpin sqlalchemy

### PR DESCRIPTION
**Pull request by prha**
## Summary & Motivation
unpin sqlalchemy 2.0.

https://github.com/dagster-io/dagster/issues/12994

## How I Tested These Changes
BK, added a `1.4` compat test suite in addition to the `1.3` test suite.

### CODE REVIEW

**Comment by prha** [comment]
Current dependencies on/for this PR:
* master
  * **PR #14480** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14480" target="_blank">Graphite</a> 
    * **PR #14481** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14481" target="_blank">Graphite</a> 
      * **PR #14482** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14482" target="_blank">Graphite</a> 
        * **PR #14483** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14483" target="_blank">Graphite</a> 
          * **PR #14484** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14484" target="_blank">Graphite</a> 
            * **PR #14485** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14485" target="_blank">Graphite</a> 
              * **PR #14486** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14486" target="_blank">Graphite</a> 
                * **PR #14500** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14500" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/14500?utm_source=stack-comment).

**Comment by prha** [comment]
Current dependencies on/for this PR:
* master
  * **PR #14480** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14480" target="_blank">Graphite</a> 
    * **PR #14481** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14481" target="_blank">Graphite</a> 
      * **PR #14482** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14482" target="_blank">Graphite</a> 
        * **PR #14483** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14483" target="_blank">Graphite</a> 
          * **PR #14484** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14484" target="_blank">Graphite</a> 
            * **PR #14485** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14485" target="_blank">Graphite</a> 
              * **PR #14486** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14486" target="_blank">Graphite</a> 
                * **PR #14500** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14500" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/14500?utm_source=stack-comment).

File: python_modules/dagster/tox.ini [hunk 1]
```diff
@@ -12,7 +12,8 @@ passenv = CI_* COVERALLS_REPO_TOKEN AWS_SECRET_ACCESS_KEY AWS_ACCESS_KEY_ID BUIL
 deps =
   scheduler_tests_old_pendulum: pendulum<2
   definitions_tests_old_pendulum: pendulum<2
-  storage_tests_old_sqlalchemy: sqlalchemy==1.3.24
+  storage_tests_sqlalchemy_1_3: sqlalchemy==1.3.24
+  storage_tests_sqlalchemy_1_4: sqlalchemy==1.4.48
```
Comment (gibsondan) [inline python_modules/dagster/tox.ini:16]: another option would be to make this sqlalchemy<1.4 and sqlalchemy<2.0 so we catch any patch updates

**Review by gibsondan — APPROVED** [review]
nice, thanks for powering through this - i'm assuming this was each of those changes from the previous PRs squashed into one, so i just took a closer look at the tox changes

**Pull request merged**